pedgrid/envs/pedestrian/PedestrianEnv.py

```diff
@@ -234,6 +234,24 @@ def _gen_grid(self, width, height):
         self.genSidewalks()
         self.mission = "switch sidewalks"
 
+    def executeObjectAction(self, action: Action):
+        if action is None:
+            return
+        
+        if action.action is ObjectAction.FORWARD:
+            self.executeForwardAction(action)
+
+    def executePositionAction(self, action: Action):
+        print ("executing position")
+        if action is None:
+            return 
+
+        agent = action.agent
+
+        logging.debug(f"position move vehicle {agent.id}")
+
+        self.positionMove(agent)
```

Comment:
positionMove is defined in the sub-class. Instead of hard-coding how an agent moves to a new location into environment, the agent should know where it wants to go (not the env). So, it's better to have a method in the agent that can produce the next position for the agent.